setup-homebrew/main.sh

```diff
@@ -0,0 +1,49 @@
+#/bin/bash
+
+set -euo pipefail
+
+if which brew &>/dev/null; then
+    HOMEBREW_PREFIX="$(brew --prefix)"
+    HOMEBREW_REPOSITORY="$HOMEBREW_PREFIX/Homebrew"
+    HOMEBREW_CORE_REPOSITORY="$HOMEBREW_REPOSITORY/Library/Taps/homebrew/homebrew-core"
+    brew update-reset "$HOMEBREW_REPOSITORY" "$HOMEBREW_CORE_REPOSITORY"
+else
+    HOMEBREW_PREFIX=/home/linuxbrew/.linuxbrew
+    HOMEBREW_REPOSITORY="$HOMEBREW_PREFIX/Homebrew"
+    sudo mkdir -p "$HOMEBREW_PREFIX"
+    sudo git clone --depth=1 https://github.com/Homebrew/brew "$HOMEBREW_REPOSITORY"
+
+    HOMEBREW_CORE_REPOSITORY="$HOMEBREW_REPOSITORY/Library/Taps/homebrew/homebrew-core"
+    sudo mkdir -p "$HOMEBREW_CORE_REPOSITORY"
+    sudo rm -rf "$HOMEBREW_CORE_REPOSITORY"
+    sudo git clone --depth=1 https://github.com/Homebrew/linuxbrew-core "$HOMEBREW_CORE_REPOSITORY"
+
+    cd "$HOMEBREW_PREFIX"
+    sudo mkdir -p bin etc include lib opt sbin share var/homebrew/linked Cellar
+    sudo ln -sf ../Homebrew/bin/brew "$HOMEBREW_PREFIX/bin/"
+    cd -
+fi
+
+export PATH="$HOMEBREW_PREFIX/bin:$PATH"
+echo "::add-path::$HOMEBREW_PREFIX/bin"
+
+GEMS_PATH="$HOMEBREW_REPOSITORY/Library/Homebrew/vendor/bundle/ruby/"
+echo "::set-output name=gems-path::$GEMS_PATH"
+GEMS_HASH=$(shasum -a 256 "$HOMEBREW_REPOSITORY/Library/Homebrew/Gemfile.lock" | cut -f1 -d' ')
+echo "::set-output name=gems-hash::$GEMS_HASH"
+
+case "$GITHUB_REPOSITORY" in
+    Homebrew/brew|Homebrew/*-core)
```

Comment:
Nice. Could make this maybe `*/brew` and similar so it could work on forks. Might want another case that does https://github.com/Homebrew/homebrew-services/blob/e68b1b8a05ee097338d2b21a0edbe1dda905c61e/.github/workflows/tests.yml#L20-L21 for all `homebrew-` taps.

Reply:
You mean do this:

```
        rm -rf /usr/local/Homebrew/Library/Taps/homebrew/homebrew-services
        ln -s "$PWD" /usr/local/Homebrew/Library/Taps/homebrew/homebrew-services
```

for all `*/homebrew-*` taps? If yes, why?

Reply:
Yes, that but with `$HOMEBREW_PREFIX/Library/Taps/$GITHUB_REPOSITORY`, if that makes sense?